File: ios/TruSdkReactNative.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TruSdkReactNative, NSObject)

RCT_EXTERN_METHOD(openCheckUrl:(NSString)url
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/TruSdkReactNative.swift


@objc(TruSdkReactNative)
class TruSdkReactNative: NSObject {

    @objc(openCheckUrl:withB:withResolver:withRejecter:)
    public func openCheckUrl(url: string, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        let trusdk: TruSDK = TruSDK()
        let response = trusdk.openCheckUrl(url, () => {
            resolve()
        })
    }
}
